### PR TITLE
Feature/upload view for csv ready to merge to main

### PR DESCRIPTION
**Pull request by trace-n**

### CODE REVIEW

File: app/controllers/expenses_controller.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    @expenses = Expense.where(tour_id:params[:id])
+
+  end
+
+  def new
+    @tour = Tour.find(params[:id])
+    @expense = Expense.new
+  end
+
+  def create
```
Comment (AwolDes) [inline app/controllers/expenses_controller.rb:16]: Do you want to keep this route for now?

File: app/controllers/expenses_controller.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      :date,
+      :city,
+      :cost,
+      :state,
+      :category,
+      :tour_id,
+    )
+
+    ap "post data paraaams"
+    ap post_data
```
Comment (AwolDes) [inline app/controllers/expenses_controller.rb:28]: These log lines can be removed

File: app/controllers/expenses_controller.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    @expense = Expense.new(post_data)
+    
+
+    if @expense.save
+
+      @url = '/tours/1/expenses'
+
+      redirect_to @url
```
Comment (AwolDes) [inline app/controllers/expenses_controller.rb:37]: Did you want to update this to be dynamic as we did in the `import` method?

File: app/models/expense.rb [hunk 4]
```diff
@@ -1,3 +1,28 @@
 class Expense < ApplicationRecord
   belongs_to :tour
+
+
+  require 'csv'
+
+  def self.import(file)
```
Comment (AwolDes) [inline app/models/expense.rb:7]: I just want to make sure you understand how this works - happy to talk about it

File: config/routes.rb [hunk 5]
```diff
@@ -22,4 +22,15 @@
   
   get "analytics" => "data_vis#data_view"
 
+  # Routes related to csv import for expenses 
+  get "tours/:id/expenses" => "expenses#index", as: "tour_expenses"
+  get "tours/:id/expenses/add" => "expenses#new"
+  get "tours/:id/expenses/import" => "expenses#show"
+  post "tours/:id/expenses" => "expenses#create"
```
Comment (AwolDes) [inline config/routes.rb:29]: Should any of these routes be deleted?

**Review by AwolDes — CHANGES_REQUESTED** [review]
The only blocking change is the `ap` lines - otherwise this is great!

File: app/controllers/expenses_controller.rb (hunk 1 above)
Comment (trace-n) [reply app/controllers/expenses_controller.rb:16]: Will keep for now - no need to reference the route

File: app/controllers/expenses_controller.rb (hunk 2 above)
Comment (AwolDes) [reply app/controllers/expenses_controller.rb:28]:
```suggestion
```

File: app/controllers/expenses_controller.rb (hunk 3 above)
Comment (trace-n) [reply app/controllers/expenses_controller.rb:35]: will leave to save time

File: app/models/expense.rb (hunk 4 above)
Comment (trace-n) [reply app/models/expense.rb:7]: thanks for the walk through

File: config/routes.rb (hunk 5 above)
Comment (trace-n) [reply config/routes.rb:29]: will leave for now

**Pull request merged**